### ISSUE TITLE
doit test_unit pyctdev RuntimeError

### ISSUE THREAD

**Issue by cdeil** [holoviz/panel#6629]
For me on MacOS I still don't manage to get a working dev env.

I followed the instructions here: https://panel.holoviz.org/developer_guide/index.html
(modulo using pip for playwright install, see #6622)

Versions: https://gist.github.com/cdeil/9291046977f5af468654af57dd083de6

Error:
```
(panel_dev) panel $ doit test_unit
Traceback (most recent call last):
  File "/Users/cdeil/anaconda3/envs/panel_dev/lib/python3.9/site-packages/doit/doit_cmd.py", line 294, in run
    return command.parse_execute(args)
  File "/Users/cdeil/anaconda3/envs/panel_dev/lib/python3.9/site-packages/doit/cmd_base.py", line 150, in parse_execute
    return self.execute(params, args)
  File "/Users/cdeil/anaconda3/envs/panel_dev/lib/python3.9/site-packages/doit/cmd_base.py", line 524, in execute
    self.loader.setup(params)
  File "/Users/cdeil/anaconda3/envs/panel_dev/lib/python3.9/site-packages/doit/cmd_base.py", line 394, in setup
    self.namespace = dict(inspect.getmembers(loader.get_module(
  File "/Users/cdeil/anaconda3/envs/panel_dev/lib/python3.9/site-packages/doit/loader.py", line 96, in get_module
    return importlib.import_module(os.path.splitext(file_name)[0])
  File "/Users/cdeil/anaconda3/envs/panel_dev/lib/python3.9/importlib/__init__.py", line 127, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 1030, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1007, in _find_and_load
  File "<frozen importlib._bootstrap>", line 986, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 680, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 850, in exec_module
  File "<frozen importlib._bootstrap>", line 228, in _call_with_frames_removed
  File "/Users/cdeil/code/oss/panel/dodo.py", line 6, in <module>
    from pyctdev import *  # noqa: api
  File "/Users/cdeil/anaconda3/envs/panel_dev/lib/python3.9/site-packages/pyctdev/__init__.py", line 44, in <module>
    from ._conda import * # noqa: api
  File "/Users/cdeil/anaconda3/envs/panel_dev/lib/python3.9/site-packages/pyctdev/_conda.py", line 55, in <module>
    from conda.models.match_spec import MatchSpec
  File "/Users/cdeil/anaconda3/envs/panel_dev/lib/python3.9/site-packages/conda/models/match_spec.py", line 18, in <module>
    from ..auxlib.collection import frozendict
  File "/Users/cdeil/anaconda3/envs/panel_dev/lib/python3.9/site-packages/conda/auxlib/collection.py", line 5, in <module>
    from .compat import isiterable
  File "/Users/cdeil/anaconda3/envs/panel_dev/lib/python3.9/site-packages/conda/auxlib/compat.py", line 8, in <module>
    deprecated.constant("24.3", "24.9", "NoneType", type(None))
  File "/Users/cdeil/anaconda3/envs/panel_dev/lib/python3.9/site-packages/conda/deprecations.py", line 256, in constant
    module, fullname = self._get_module(stack)
  File "/Users/cdeil/anaconda3/envs/panel_dev/lib/python3.9/site-packages/conda/deprecations.py", line 335, in _get_module
    for module in sys.modules.values():
RuntimeError: dictionary changed size during iteration
```

**Comment by cdeil** [comment]
When I simply run pytest I get this collection error.

What's the proper thing to do here? Add `__init__.py` file to make `scripts` a proper package to make the imports work?
Exclude `scripts` from default collect path (even though it has pytests)?

```
(panel_dev) panel $ pytest          
==================================================================================== test session starts ====================================================================================
platform darwin -- Python 3.9.19, pytest-7.4.4, pluggy-1.4.0 -- /Users/cdeil/anaconda3/envs/panel_dev/bin/python3.9
cachedir: .pytest_cache
rootdir: /Users/cdeil/code/oss/panel
configfile: pyproject.toml
plugins: cov-5.0.0, anyio-4.3.0, rerunfailures-14.0, asyncio-0.21.1, playwright-0.4.4, xdist-3.5.0, base-url-2.1.0, nbval-0.11.0
asyncio: mode=auto
collected 3452 items / 2 errors / 1080 deselected / 2 skipped / 2372 selected                                                                                                               

========================================================================================== ERRORS ===========================================================================================
__________________________________________________________________ ERROR collecting scripts/panelite/test/test_panelite.py __________________________________________________________________
ImportError while importing test module '/Users/cdeil/code/oss/panel/scripts/panelite/test/test_panelite.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
../../../anaconda3/envs/panel_dev/lib/python3.9/importlib/__init__.py:127: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
scripts/panelite/test/test_panelite.py:45: in <module>
    from scripts.test_panelite.notebooks_with_panelite_issues import (
E   ModuleNotFoundError: No module named 'scripts.test_panelite'
___________________________________________________________________ ERROR collecting scripts/panelite/test/test_utils.py ____________________________________________________________________
ImportError while importing test module '/Users/cdeil/code/oss/panel/scripts/panelite/test/test_utils.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
../../../anaconda3/envs/panel_dev/lib/python3.9/importlib/__init__.py:127: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
scripts/panelite/test/test_utils.py:3: in <module>
    from scripts.test_panelite.utils import MaxRetriesExceeded, Retrier
E   ModuleNotFoundError: No module named 'scripts.test_panelite'
================================================================================== short test summary info ==================================================================================
ERROR scripts/panelite/test/test_panelite.py
ERROR scripts/panelite/test/test_utils.py
!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!! Interrupted: 2 errors during collection !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
======================================================================= 2 skipped, 1080 deselected, 2 errors in 5.48s =======================================================================
```

**Comment by Hoxbro** [comment]
For 1 downgrade conda to below 24.3 in your panel environment. 

For 2 run `pytest panel` instead

**Comment by cdeil** [comment]
@Hoxbro - Indeed this fixed it, thanks!

```
conda install 'conda<24.3'
```

I can now run the tests and see only a few fails: #6635

**Comment by cdeil** [comment]
Fixed by changing to pixi in #6775
